stdlib/public/Concurrency/Task.cpp

```diff
@@ -503,32 +530,93 @@ static AsyncTaskAndContext swift_task_create_commonImpl(
+    SWIFT_TASK_DEBUG_LOG("New detached task base pri = %zu", basePriority);
+
+  } else if (swift_task_is_unstructured(jobFlags)) {
+
+    if (task_priority_is_unspecified(basePriority)) {
+      // Case 1: No priority specified
+      //    Base priority = Base priority of parent with a UI -> IN downgrade
+      //    Escalated priority = UN
+      if (currentTask) {
+        basePriority = currentTask->_private().BasePriority;
+        SWIFT_TASK_DEBUG_LOG("Creating unstructured task from current_task %p at %zu", currentTask, basePriority);
+      } else {
+        basePriority = swift_task_getCurrentThreadPriority();
+        SWIFT_TASK_DEBUG_LOG("Creating unstructured task from current thread at %zu", basePriority);
+      }
+      basePriority = withPriorityDowngrade(basePriority);
+      SWIFT_TASK_DEBUG_LOG("New unstructured task base pri = %zu", basePriority);
+    } else {
+      // Case 2: User specified a priority
+      //    Base priority = user specified priority
+      //    Escalated priority = UN
+      SWIFT_TASK_DEBUG_LOG("New unstructured task with user specified base pri = %zu", basePriority);
+    }
+
+    // Task will be created with max priority = max(base priority, UN) = base
+    // priority
+  } else {
+    // Is a structured concurrency child task. Must have a parent.
+    assert((asyncLet || group) && parent);
+
+    if (basePriority == JobPriority::Unspecified) {
+      // Case 1: No priority specified
+      //    Base priority = Base priority of parent with a UI -> IN downgrade
+      //    Escalated priority = Escalated priority of parent with a UI -> IN
+      //    downgrade
+      JobPriority parentBasePri = parent->_private().BasePriority;
+      SWIFT_TASK_DEBUG_LOG("Creating structured task from parent %p at %zu", parent, parentBasePri);
+      basePriority = withPriorityDowngrade(parentBasePri);
+      SWIFT_TASK_DEBUG_LOG("New structured task with propagated base pri = %zu", basePriority);
+    } else {
+      // Case 2: User priority specified
+      //    Base priority = User specified priority
+      //    Escalated priority = Escalated priority of parent with a UI -> IN
+      //    downgrade
+      SWIFT_TASK_DEBUG_LOG("Creating structured task from parent %p", parent);
+      SWIFT_TASK_DEBUG_LOG("New structured task with user specified base pri = %zu", basePriority);
+    }
+
+    // Task will be created with escalated priority = base priority. We will
+    // update the escalated priority with the right rules in
+    // updateNewChildWithParentAndGroupState when we link the child into
+    // the parent task/task group since we'll have the right
+    // synchronization then.
   }
 
-  // Adjust user-interactive priorities down to user-initiated.
-  if (jobFlags.getPriority() == JobPriority::UserInteractive)
-    jobFlags.setPriority(JobPriority::UserInitiated);
+  // TODO (rokhinip): This shouldn't be needed, why is this needed to make
```

Comment:
This is more of a note to self, I omitted this during my initial PR review and found some tests behaving strangely in a way I didn't fully understand. I need to explore this further.